shared/src/main/scala/effekt/Parser.scala

```diff
@@ -244,7 +244,11 @@ class Parser(positions: Positions) extends Parsers(positions) with Phase[Source,
     )
 
   lazy val effectOp: P[Operation] =
-    idDef ~ maybeTypeParams ~ some(valueParams) ~/ (`:` ~/> valueType) ^^ Operation
+    idDef ~ maybeTypeParams ~ some(valueParams) ~/ (`:` ~/> effectful) ^^ {
+      case id ~ tparams ~ params ~ ret =>
+        val isEff = ret.eff.effs.size > 0
```

Comment:
Maybe instead of computing it here and storing it as a field, you could add it as a method on Operation?.

Reply:
It is not used, so I removed it.